### PR TITLE
Flesh out explore page

### PR DESCRIPTION
**Pull request by divad12**
Explore page:

flow-android-explore-courses

Pressing ECON 102:

flow-android-econ-102

Note about change in API code: all instances of `search*` in method/constant names has been replaced with `get*`, since we actually have an API endpoint corresponding to search now.

Still TODO:
- [ ] Actually use UI state to pass search params
- [ ] Spinner/gray-out while results are loading
- [ ] Spinner/gray-out when course page is loading

### CODE REVIEW

File: res/layout/explore_layout.xml [hunk 1]
```diff
@@ -19,20 +19,22 @@
...
+                  android:inputType="text"
                   android:layout_width="0dp"
                   android:layout_height="wrap_content"
                   android:layout_weight="1"
                   android:textSize="22dp"
                   android:singleLine="true"
                   android:textColor="@color/white"
-                  android:text="e.g. econ101, psych"
-                />
+                  android:textColorHint="@color/grey_light"
```
Comment (jasperfung) [inline res/layout/explore_layout.xml]: Can we go lighter? Clashes with the blue a lot.

Comment (divad12) [inline res/layout/explore_layout.xml]: K, I'll try #EEE or just white.

File: res/layout/explore_layout.xml [hunk 2]
```diff
@@ -19,20 +19,22 @@
...
+                  android:hint="eg. econ101, cs"/>
         <Button android:id="@+id/temp_search_button"
```
Comment (jasperfung) [inline res/layout/explore_layout.xml]: Let's get rid of this button if we can use the keyboard to trigger a search.

Comment (divad12) [inline res/layout/explore_layout.xml]: Sounds good.

**Comment by divad12** [comment]
@jasperfung Thanks for the review. Comments addressed.

**Comment by jasperfung** [comment]
LGTMerge

**Pull request merged**